### PR TITLE
Implementation of HTTPD + FPM use case

### LINKED ISSUE #1: Restructure buildpack: PHP Start for HTTPD
## Context
[PHP RFC  0003](https://github.com/paketo-buildpacks/rfcs/blob/d4fde0656a823330f78a77e69a81d2d6f7768daf/text/php/0003-php-start.md) is an addendum to [PHP Restructure RFC 0001](https://github.com/paketo-buildpacks/rfcs/blob/d4fde0656a823330f78a77e69a81d2d6f7768daf/text/php/0001-restructure.md
). The RFC outlines creating a PHP Start buildpack for managing the task of running start commands for HTTPD, Nginx, and FPM.

## Issue
This is to add the new PHP start buildpack base implementation with support for the HTTPD + FPM start command use case. Separate issues will be filed to address Nginx + FPM, HTTPD only, and Nginx only.

Logic to support running two start commands already lives in the PHP Web buildpack `procmgr`, and a similar implementation can be used in this buildpack.

- [x] API: the buildpack should require: `php`, `php-fpm` (optional), and `httpd-config` at build, and
    `php-fpm` (optional), `httpd`, and `httpd-config` and launch.
- [x] Buildpack API Version: This buildpack should be written using [Buildpack API 0.7](https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.7)
- [x] README: Make sure a README is added that conveys context about the buildpack in a manner consistent with other Paketo buildpacks.
- [x] Use direct process types where applicable

## Acceptance
I can successfully build an PHP app with a HTTPD server using the Paketo PHP Dist Buildpack, Paketo PHP FPM buildpack, the Paketo PHP HTTPD buildpack, and this new buildpack, specifying an intent to use HTTPD as the server of choice. I should be able to ping the app successfully, and container logs show that FPM is running inside the container as well.

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Resolves #1 

~~**BLOCKED** on the merge and release of:~~
~~https://github.com/paketo-buildpacks/php-fpm/pull/4~~
~~https://github.com/paketo-buildpacks/php-httpd/pull/3~~

## Use Cases

Supports HTTPD and FPM running in the same container

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: build_test.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+					Plan: packit.BuildpackPlan{
+						Entries: []packit.BuildpackPlanEntry{},
+					},
+					Layers: packit.Layers{Path: layersDir},
+				})
+				Expect(err).To(MatchError(ContainSubstring("failed to parse layer content metadata:")))
+			})
+		})
+
+		context("when the python layer cannot be reset", func() {
```
Comment (thitch97) [inline build_test.go:199]:
```suggestion
		context("when the php-start layer cannot be reset", func() {
```

File: build_test.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+		build = phpstart.Build(procMgr, logEmitter)
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(layersDir)).To(Succeed())
+		Expect(os.RemoveAll(cnbDir)).To(Succeed())
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
+	})
+
+	context("the PHP_HTTPD_PATH env var is set", func() {
```
Comment (thitch97) [inline build_test.go:60]: I think I understand why this case is here, but just to validate my own understanding:
Am I correct in thinking this case would be an unlikely one given `PHPRC`, `PHP_FPM_PATH` and `PHP_HTTPD_PATH` would have been by previous buildpacks?

Comment (sophiewigmore) [reply build_test.go:60]: Yes, you're right on. This won't happen as of now with the way the buildpacks are set up. The plan is to eventually (post re-write probably) support FPM running in its own container, in which case this situation will become more common.

Comment (thitch97) [reply build_test.go:60]: Understood. The `php-fpm` buildpack is expected to be optional based on [RFC 0003](https://github.com/paketo-buildpacks/rfcs/blob/f79658839d4fe3fb3f4795f68cd63e8b8f094d70/text/php/0003-php-start.md?plain=1#L92), but it doesn't appear to be in this case. Is that also related to the plan stated above?

**Review by thitch97 — APPROVED** [review]

**Pull request merged**